http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -1410,16 +1410,39 @@ class LeoPubsubMessageSubscriber[F[_]](
             .updateAndPollApp(msg.appId, msg.appName, latestAppChartVersion, msg.workspaceId, azureContext)
       }
 
-      _ <- updateApp.adaptError { case e =>
-        PubsubKubernetesError(
-          AppError(e.getMessage, ctx.now, ErrorAction.UpdateApp, ErrorSource.App, None, Some(ctx.traceId)),
-          Some(msg.appId),
-          false,
-          None,
-          None,
-          None
-        )
-      }
+      _ <- updateApp
+        .handleErrorWith {
+          // Fatal case, polling app liveliness failed
+          // The two fatal cases are included separately, because later we may wish to fail fatally on `HelmException`, but roll back on `AppUpdatePollingException`
```

Comment:
By fatal here, you mean that the app will transition from `Updating` to `Error` correct?